nixos/modules/services/web-servers/ttyd.nix

```diff
@@ -0,0 +1,202 @@
+         ++ optional cfg.checkOrigin "--check-origin"
+         ++ optional (cfg.maxClients != null) "--max-clients ${cfg.maxClients}"
+         ++ optional (cfg.indexFile != null) "--index ${cfg.indexFile}"
+         ++ optional cfg.enableIPv6 "--ipv6"
+         ++ optional cfg.enableSSL "--ssl-cert ${cfg.certFile} --ssl-key ${cfg.keyFile} --ssl-ca ${cfg.caFile}"
+         ++ optional (cfg.logLevel != null) "--debug ${toString cfg.logLevel}"
+
+  ;
+
+  # Command to start ttyd
+  cmd = "${pkgs.ttyd}/bin/ttyd ${concatStringsSep " " args} ${pkgs.shadow}/bin/login";
+
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+    services.ttyd = {
+      enable = mkEnableOption "ttyd daemon";
+
+      port = mkOption {
+        type = types.int;
+        default = 7681;
+        description = "Port to listen on (default: 7681, use 0 for random port)";
+      };
+
+      socket = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/run/ttyd.sock";
+        description = "UNIX domain socket path to bind (eg: /var/run/ttyd.sock)";
+      };
+
+      interface = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "eth0";
+        description = "Network interface to bind (eg: eth0)";
+      };
+
+      username = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Username for basic authentication.";
+      };
+
+      password = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Password for basic authentication.
+          Warning: this is stored in cleartext in the Nix store!
+        '';
+      };
+
+      uid = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "User id to run with.";
+      };
+
+      gid = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Group id to run with.";
+      };
+
+      signal = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        description = "Signal to send to the command when exit it (default: 1, SIGHUP).";
+      };
+
+      clientOptions = mkOption {
+        type = types.attrsOf types.str;
+        default = {};
+        example = {
+          "fontSize" = "16";
+          "fontFamily" = "Fira Code";
+        };
+        description = ''
+          Attribute set of client options for xtermjs.
+          https://xtermjs.org/docs/api/terminal/interfaces/iterminaloptions/
+        '';
+      };
+
+      terminalType = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "xterm-256color";
+        description = "Terminal type to report, default: xterm-256color.";
+      };
+
+      checkOrigin = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Do not allow websocket connection from different origin.";
```

Comment:
I'd use "Whether to allow a websocket connection from a different origin."

Reply:
Done. Copied the blurb straight from the command line help.